tests/framework/utils.py

```diff
@@ -760,17 +761,29 @@ def start_screen_process(screen_log, session_name, binary_path, binary_params):
         delay=1,
     ).group(1)
 
-    binary_clone_pid = int(
-        open("/proc/{0}/task/{0}/children".format(screen_pid), encoding="utf-8")
-        .read()
-        .strip()
-    )
+    # Make sure the screen process launched successfully
+    # As the parent process for the binary.
+    screen_ps = psutil.Process(int(screen_pid))
+    for _ in range(5):
```

Comment:
Why 5? Since we are using the pid as a filter, we expect to only have 1 entry here always, right?

Reply:
Yea, the loop here is just a retry of `screen_ps.is_running()` until the process reaches a running state. The reason the buildkite job crashed is because the `/proc/{}/task/{}/children` file existed but was empty. The retry here is to make sure the screen process process is up and running before checking for the child process.

Reply:
ah ok. For cases like this one, we have a heavy use in the python scripts for retry calls.

You can see here a bunch of them: https://github.com/firecracker-microvm/firecracker/blob/main/tests/framework/microvm.py#L499

You would want something like this:
```
@retry(delay=0.5, tries=5)
def wait_screen_ps(screen_ps):
  assert screen_ps.is_running()
```

Reply:
That is cleaner, thanks!